pkg/endpoint/bpf.go

```diff
@@ -145,11 +145,11 @@ func (e *Endpoint) writeHeaderfile(prefix string) error {
 }
 
 // addNewRedirectsFromDesiredPolicy must be called while holding the endpoint lock for
-// writing. On success, returns nil; otherwise, returns an error  indicating the
+// writing. On success, returns nil; otherwise, returns an error indicating the
 // problem that occurred while adding an l7 redirect for the specified policy.
 // Must be called with endpoint.Mutex held.
 func (e *Endpoint) addNewRedirectsFromDesiredPolicy(ingress bool, desiredRedirects map[string]bool, proxyWaitGroup *completion.WaitGroup) (error, revert.FinalizeFunc, revert.RevertFunc) {
-	if option.Config.DryMode {
+	if option.Config.DryMode || e.isProxyDisabled() {
```

Comment:
Not strictly, but instead of refactoring `updateProxyRedirect()` which was only called from here, I inlined it here instead. `e.isProxyDisabled()` came from there.